packages/@react-spectrum/toast/stories/Toast.stories.tsx

```diff
@@ -146,3 +153,99 @@ function MultipleInner() {
     </>
   );
 }
+
+function IframeExample() {
+  let onLoad = (e: SyntheticEvent) => {
+    let iframe = e.target as HTMLIFrameElement;
+    let window = iframe.contentWindow;
+    let document = window.document;
+
+    // Catch toasts inside the iframe and redirect them outside.
+    window.addEventListener('react-spectrum-toast', (e: CustomEvent) => {
+      e.preventDefault();
+      ToastQueue[e.detail.variant](e.detail.children, e.detail.options);
+    });
+
+    let prevFocusedElement = null;
+    window.addEventListener('react-aria-landmark-navigation', (e: CustomEvent) => {
+      e.preventDefault();
+      let el = document.activeElement;
+      if (el !== document.body) {
+        prevFocusedElement = el;
+      }
+
+      // Prevent focus scope from stealing focus back when we move focus to the iframe.
+      document.body.setAttribute('data-react-aria-top-layer', 'true');
+
+      window.parent.postMessage({
+        type: 'landmark-navigation',
+        direction: e.detail.direction
+      });
+
+      setTimeout(() => {
+        document.body.removeAttribute('data-react-aria-top-layer');
+      }, 100);
+    });
+
+    // When the iframe is re-focused, restore focus back inside where it was before.
+    window.addEventListener('focus', () => {
+      if (prevFocusedElement) {
+        prevFocusedElement.focus();
+        prevFocusedElement = null;
+      }
+    });
+
+    // Move focus to first or last landmark when we receive a message from the parent page.
+    window.addEventListener('message', e => {
```

Comment:
Just a story thing I noticed: if you move out of the dialog in the iframe and onto the toast via F6, you can't get back inside the iframe. Maybe this is because the `main` is aria-hidden due to the dialog being open?

Reply:
Yep that's exactly it. There are no landmarks visible in the iframe anymore so landmark navigation skips over them.